### PR TITLE
refactor(web): deployment http api

### PR DESCRIPTION
**Pull request by m8vago**
Add deployment http api.
Remove deployment grpc api.
Fix http error handling.

### CODE REVIEW

File: web/crux/src/app/deploy/deploy.dto.ts [hunk 1]
```diff
@@ -1,26 +1,142 @@
...
-  @IsEnum(DeploymentStatusDto)
+  @ApiProperty({ enum: DEPLOYMENT_STATUS_VALUES })
+  @IsIn(DEPLOYMENT_STATUS_VALUES)
   status: DeploymentStatusDto
+}
+
+export class DeploymentDto extends BasicDeploymentDto {
+  @IsString()
+  @IsOptional()
+  note?: string
```
Comment (polaroi8d) [inline web/crux/src/app/deploy/deploy.dto.ts:26]: Don't need the `@IsOptional()` if there is the question mark. Same as above.

File: web/crux/src/app/deploy/deploy.dto.ts [hunk 2]
```diff
@@ -1,26 +1,142 @@
...
 
+  @Type(() => Date)
+  @IsDate()
   updatedAt: Date
 
   node: BasicNodeDto
 }
+
+export class InstanceContainerConfigDto extends OmitType(PartialType(ContainerConfigDto), ['secrets']) {
+  secrets?: UniqueSecretKeyValue[] | null
```
Comment (polaroi8d) [inline web/crux/src/app/deploy/deploy.dto.ts:50]: Why need the `| null` if the field is optional?

File: web/crux/src/app/deploy/deploy.http.controller.ts [hunk 3]
```diff
@@ -1,52 +1,174 @@
...
+const InstanceId = () => Param('instanceId')
 
-@UseGuards(JwtAuthGuard)
-@UseInterceptors(HttpLoggerInterceptor, PrismaErrorInterceptor)
-@UseFilters(HttpExceptionFilter)
-@Controller('deploy')
+const ROUTE_DEPLOYMENTS = 'deployments'
+const ROUTE_DEPLOYMENT_ID = ':deploymentId'
+const ROUTE_INSTANCES = 'instances'
+const ROUTE_INSTANCE_ID = ':instanceId'
```
Comment (polaroi8d) [inline web/crux/src/app/deploy/deploy.http.controller.ts:48]: Am I the only one who is missing a new line here?

File: web/crux/src/app/deploy/deploy.http.controller.ts [hunk 4]
```diff
@@ -1,52 +1,174 @@
...
+    return await this.service.getDeploymentEvents(deploymentId)
+  }
+
+  @Get(`${ROUTE_DEPLOYMENT_ID}/${ROUTE_INSTANCES}/${ROUTE_INSTANCE_ID}`)
+  @ApiOkResponse({ type: InstanceDto })
+  async getInstance(@InstanceId() instanceId: string): Promise<InstanceDto> {
+    return await this.service.getInstance(instanceId)
+  }
+
+  @Get(`${ROUTE_DEPLOYMENT_ID}/${ROUTE_INSTANCES}/${ROUTE_INSTANCE_ID}/secrets`)
```
Comment (polaroi8d) [inline web/crux/src/app/deploy/deploy.http.controller.ts:92]: I love the way how we use constants, but right here when I check this controller I will never be able to figure out what path this is. Just a comment.

File: web/crux/src/app/version/version.dto.ts [hunk 5]
```diff
@@ -1,27 +1,9 @@
...
-  name: string
-
-  @ApiProperty({
-    enum: VERSION_TYPE_VALUES,
-  })
-  @IsString()
-  @IsIn(VERSION_TYPE_VALUES)
-  type: VersionTypeDto
-}
+import { AuditDto, BasicVersionDto, VersionTypeDto, VERSION_TYPE_VALUES } from '../shared/shared.dto'
```
Comment (polaroi8d) [inline web/crux/src/app/version/version.dto.ts:6]: :(

**Review by polaroi8d — CHANGES_REQUESTED** [review]

File: web/crux/src/app/deploy/deploy.dto.ts (hunk 2 above)
Comment (m8vago) [reply web/crux/src/app/deploy/deploy.dto.ts:50]: https://upmostly.com/angular/optional-vs-nullable-in-typescript-angular

File: web/crux/src/app/deploy/deploy.http.controller.ts (hunk 3 above)
Comment (m8vago) [reply web/crux/src/app/deploy/deploy.http.controller.ts:48]: No, me too, just the gitlense got me confused. nice catch. :D

File: web/crux/src/app/shared/shared.dto.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  @IsUUID()
+  createdBy: string
+
+  @Type(() => Date)
+  @IsDate()
+  updatedAt: Date
+
+  @IsUUID()
+  updatedBy?: string
+}
```
[the remark this answers is not in the document]
Comment (m8vago) [reply web/crux/src/app/shared/shared.dto.ts:30]: gitlense :(

**Review by robot9706 — APPROVED** [review]

**Pull request merged**